Release/0.1.0

adatzer opened the pull request.

paulboocock commented on metadata.yaml:
```diff
@@ -0,0 +1,7 @@
+documentation: "https://docs.snowplowanalytics.com/docs"
+versions:
+  # Latest version
+  - sha: 32ce1443a58107a7dc3b01b30cd5f5a1cd2e17a0
```
Just a reminder to merge commit this into `main` so we don't change the hash with a rebase and merge.

paulboocock commented on template.tpl:
```diff
@@ -0,0 +1,1260 @@
+  "id": "cvt_temp_public_id",
+  "version": 1,
+  "securityGroups": [],
+  "displayName": "Iterable HTTP API v1",
```
```suggestion
  "displayName": "Iterable",
```

I only put v2 in the Amplitude tag because Amplitude explictly call this integration HTTP API v2. I don't think iterable really do the same, so I'm tempted to just call this Iterable. Especially as it does user updating and even tracking, rather than only implement a single API.

paulboocock commented on template.tpl:
```diff
@@ -0,0 +1,1260 @@
+    "displayName": "",
+    "thumbnail": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAQ0AAAD/CAYAAADrP4OuAAAwEnpUWHRSYXcgcHJvZmlsZSB0eXBlIGV4aWYAAHjatZxpdh03sq3/YxQ1BPTNcNCudWfwhn+/jTyHIiXKlp/rlsoWTR5mIoGI3QQCafb/+59j/vOf/zhXkzUxlZpbzpb/xRab73xR7fO/dv/tbLz/fv63Xl+5r98357w+4PlW4O/w/Gfpz9+u8/304xfe93Dj6/dNff3E19eF3nd+XTDozv6O5NMg+b5/vu/i60JtP1/kVsvnoQ7//D1fH7xDef1zpr/XS+P5kf7bfP5GLMzSStwoeL+DC/b+Oz4jCPrHhc7f+f6bQekzfB1 ...
+  },
+  "description": "Tag that sends events to Iterable from Snowplow, GAv4 or other clients to Iterable HTTP API v1. Works best with the Snowplow Client.",
```
```suggestion
  "description": "Tag that sends events to Iterable from Snowplow, GAv4 or other clients to Iterable. Works best with the Snowplow Client.",
```

paulboocock commented on template.tpl:
```diff
@@ -0,0 +1,1260 @@
+  return undefined;
+};
+
+const isIdentityEvent = (eventData, tagConfig) => {
+  if (tagConfig.useDefaultIdentify) {
+    return eventData.event_name === 'identify';
+  }
+
+  const identityEvents = tagConfig.identityEventsByName;
+  if (getType(identityEvents) === 'array') {
+    return identityEvents.indexOf(eventData.event_name) >= 0;
+  }
+
+  return false;
+};
```
👍

paulboocock commented on template.tpl:
```diff
@@ -0,0 +1,1260 @@
+  };
+  const userIdentifiers = {
+    email: locate(idLocations.email),
+    userId: locate(idLocations.userId)
```
This is a rather nice way of solving this.

paulboocock commented on template.tpl:
```diff
@@ -0,0 +1,1260 @@
+  sendHttpRequest(
+    iterableEndpoints.eventsTrack,
+    (statusCode, headers, body) => {
+      if (statusCode >=200 && statusCode < 300) {
```
nit:
```suggestion
      if (statusCode >= 200 && statusCode < 300) {
```

It might be worth running this through a default Prettier for a formatting run. https://prettier.io/docs/en/cli.html

paulboocock reviewed (approved):
LGTM.

I think we've dealt with the Identify problem well. We'll have to ensure it's well documented to help people avoid 400's on every track event but I think that resending with the placeholder is a nice way to work with this.

adatzer commented:
Thank you for reviewing @paulboocock !
Changes force-pushed:
- Renamed from `Iterable HTTP API v1` to just `Iterable`.
- Ran `prettier --single-quote`
- Changed the latest sha to the new one in `metadata.yaml`

Indeed the identity settings need to be well documented too. For reference, concerning the creation of the placeholder email, this is what Iterable does too with the `preferUserId: true` option. The reason we create it instead of using that option is to make it predictable (`{{userId}}@placeholder.email`), which could be useful for an Iterable user that might want to later use the `updateEmail` API.
